### ISSUE TITLE
Automatically utilizing the username and password from the FTP address.

### ISSUE THREAD

**Issue by Vixb1122** [files-community/Files#14953]
### What feature or improvement do you think would benefit Files?

When opening an FTP address, it is unable to automatically utilize the username and password provided within the address as credentials.

### Requirements

1. Add a ftp url to as a Network Drive
2. Url is 'ftp://root:pwd123@11.10.119.33/'
3. Automatically utilizing the username and password

### Files Version

3.2.2.0

### Windows Version

22631.3296

**Comment by Josh65-2201** [comment]
Thanks for the feedback, I've added this to the project board.

**Comment by btomblinson** [comment]
I am trying to test this out, the FTP url isn't working for me

**Comment by yaira2** [comment]
> I am trying to test this out, the FTP url isn't working for me

It's just an example, you would have to setup something similar on your end.

**Comment by btomblinson** [comment]
Yeah, I created a quick Linux VM on AWS and put FTP on it, I was able to connect from FileZilla, I think it needs to be on local network maybe, because I think its expecting a \\<servername> format

**Comment by btomblinson** [comment]
@Josh65-2201 can you confirm that example URL or similar works for you if you have a local FTP server?

**Comment by Josh65-2201** [comment]
I can connect to an FTP server fine in Files using the IP only or same format

**Comment by btomblinson** [comment]
Can you provide a quick video of your steps?

**Comment by btomblinson** [comment]
@Josh65-2201 I tried this out and it does work, but from my testing Files does not even save this shortcut

https://github.com/files-community/Files/assets/16448374/17fb8819-4e81-4b42-acfc-2c08d58e1da9

**Comment by btomblinson** [comment]
To note, it does open it, and it does use the user and password from the URL, it just doesn't seem to save the shortcut

**Comment by Josh65-2201** [comment]
Have you tried doing/seeing if it's there in Windows File Explorer?

**Comment by btomblinson** [comment]
Yes it is but not this application

**Comment by om2805** [comment]
Hi, I’d like to take this issue if it’s still open. I plan to extract credentials from the FTP URL and reuse the existing authentication flow without showing the dialog. Please let me know if that approach works

**Comment by yair100** [comment]
@om2805 that sounds reasonable. Does anyone have a different idea?

**Comment by Shamveel-Khan** [comment]
I'd like to work on this issue. I plan to extract the credentials from the FTP URL and reuse the existing authentication flow, so the username and password are automatically used without showing the authentication dialog.
